containerd-shim/process_solaris.go

```diff
@@ -0,0 +1,62 @@
+// +build solaris
+
+package main
+
+import (
+	"io"
+	"os"
+	"syscall"
+)
+
+// setPDeathSig is a no-op on Solaris as Pdeathsig is not defined.
+func setPDeathSig() *syscall.SysProcAttr {
+	return nil
+}
+
+// openIO opens the pre-created fifo's for use with the container
+// in RDWR so that they remain open if the other side stops listening
+func (p *process) openIO() error {
+	p.stdio = &stdio{}
+	var (
+		uid = p.state.RootUID
+	)
+	i, err := p.initializeIO(uid)
+	if err != nil {
+		return err
+	}
+	p.shimIO = i
+	// non-tty
```

Comment:
Oh, I see.

I got confused because of `// non-tty` comment.

Maybe just replace that comment with something akin to `// both tty and non-tty mode are handled by the runtime using the following pipes` ?